genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobKickoffTask.java

```diff
@@ -222,14 +224,15 @@ protected synchronized void createUser(final String user, final String group) th
             userCreateCommandLine.addArgument("useradd");
             userCreateCommandLine.addArgument(user);
 
-            if (StringUtils.isNotBlank(group)) {
+            if (StringUtils.isNotBlank(group) && !group.equals(user)) {
```

Comment:
Since this check seems equivalent to the one on line 207 why don't you create a boolean and assign it to the result of these statements so that you don't have to run the method calls twice?